redis/src/main/java/org/hibernate/ogm/datastore/redis/RedisDialect.java

```diff
@@ -0,0 +1,546 @@
+					if ( value != null ) {
+						row.put( columnName, value );
+					}
+				}
+
+				rows.add( row );
+			}
+		}
+		return rows;
+	}
+
+	@Override
+	public void removeAssociation(AssociationKey key, AssociationContext associationContext) {
+		if ( isStoredInEntityStructure( key.getMetadata(), associationContext.getAssociationTypeContext() ) ) {
+			Entity owningEntity = getEntity(
+					key.getEntityKey(),
+					associationContext
+			);
+			if ( owningEntity != null ) {
+				owningEntity.removeAssociation( key.getMetadata().getCollectionRole() );
+				storeEntity( key.getEntityKey(), owningEntity, associationContext );
+			}
+		}
+		else {
+			removeAssociation( key.getEntityKey() );
+		}
+	}
+
+	public Entity getEntity(EntityKey key, OptionsContext optionsContext) {
+		return getEntityStorageStrategy( optionsContext ).getEntity( entityId( key ) );
+	}
+
+	public void storeEntity(EntityKey key, Map<String, Object> map, OptionsContext optionsContext) {
+		Entity entityDocument = new Entity();
+		Set<String> keys = new HashSet<>( Arrays.asList( key.getColumnNames() ) );
+
+
+		for ( Map.Entry<String, Object> entry : map.entrySet() ) {
+			if ( keys.contains( entry.getKey() ) ) {
+				continue;
+			}
+			entityDocument.set( entry.getKey(), entry.getValue() );
+		}
+
+
+		storeEntity( key, entityDocument, optionsContext );
+
+		Long ttl = getTTL( optionsContext );
+		if ( ttl != null ) {
+			expireEntity( key, ttl );
+		}
+	}
+
+	public void storeEntity(EntityKey key, Entity document, OptionsContext optionsContext) {
+		getEntityStorageStrategy( optionsContext ).storeEntity( entityId( key ), document );
+	}
+
+	public Association getAssociation(EntityKey key) {
+
+		byte[] associationId = associationId( key );
+		List<byte[]> lrange = connection.lrange( associationId, 0, -1 );
+
+		Association association = new Association();
+
+		for ( byte[] bytes : lrange ) {
+			association.getRows().add( serializationStrategy.deserialize( bytes, Object.class ) );
+		}
+		return association;
+	}
+
+	public Entity getEntity(EntityKey key, AssociationContext associationTypeContext) {
+		return getEntityStorageStrategy(
+				associationTypeContext.getAssociationTypeContext()
+						.getOptionsContext()
+		).getEntity( entityId( key ) );
+	}
+
+	public Entity storeEntity(EntityKey key, Entity entity, AssociationContext associationContext) {
+		getEntityStorageStrategy( associationContext.getAssociationTypeContext().getOptionsContext() ).storeEntity(
+				entityId( key ),
+				entity
+		);
+		return entity;
+	}
+
+	private void expireEntity(EntityKey key, Long ttl) {
+		byte[] associationId = entityId( key );
+		connection.pexpire( associationId, ttl );
+	}
+
+	public void storeAssociation(EntityKey key, Association document) {
+
+		byte[] associationId = associationId( key );
+		connection.del( associationId );
+
+		for ( Object row : document.getRows() ) {
+			connection.rpush( associationId, serializationStrategy.serialize( row ) );
```

Comment:
Result of our discussion: We'll start without transactions in the first step. Transactions require multiple connections/connection pooling. Pooling and transactions will be done in a latter step.